fabfile.py

```diff
@@ -0,0 +1,206 @@
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+from fabric.decorators import task
+from fabric.tasks import Task
+from fabric.api import run
+from fabric.api import cd
+from fabric.api import execute
+from fabric.contrib.files import exists
+from fabric.context_managers import settings
+from utils import arch, open_dir, clone
+from utils import get_list_of_failed_test_cases
+from utils import get_known_issues
+from utils import filter_known_issues
+import config as config
+
+
+class TaskWithConfig(Task):
+    def __init__(self, func, *args, **kwargs):
+        super(TaskWithConfig, self).__init__(*args, **kwargs)
+        self.func = func
+
+    def run(self, custom_config=None, *args, **kwargs):
+        if custom_config is not None:
+            config.set_custom_config(custom_config)
+        return self.func(*args, **kwargs)
+
+
+@task(task_class=TaskWithConfig)
+def clone_sdl(rewrite=False):
+    """
+    Clone SDL to work directory
+    :param rewrite: if True delete sdl folder
+    """
+    clone(config.sdl_repository, config.sdl_clone_dir, config.work_dir, branch=config.sdl_branch, rewrite=rewrite)
+
+
+@task(task_class=TaskWithConfig)
+def clone_atf(rewrite=False):
+    """
+    Clone ATF to work directory
+    :param rewrite: if True delete atf folder
+    """
+    clone(config.atf_repository, config.atf_build_dir, config.work_dir, branch=config.atf_branch, submodules=True, rewrite=rewrite)
+
+
+@task(task_class=TaskWithConfig)
+def clone_scripts(rewrite=False):
+    """
+    Clone ATF scripts to work directory
+    :param rewrite: if True delete atf scripts folder
+    """
+    clone(config.scripts_repository, config.scripts_clone_dir, config.work_dir, branch=config.scripts_branch,
+          rewrite=rewrite)
+
+
+@task(task_class=TaskWithConfig)
+def build_sdl(rewrite=False):
+    """
+    Build SDL
+    :param rewrite: if True delete old SDL build and build again
+    """
+    with open_dir(config.work_dir), open_dir(config.sdl_build_dir, rewrite=rewrite):
+        log_build_dir = config.sdl_build_dir + "log4cxx_build/"
+        log_build_arch_dir = log_build_dir + arch()
+        run('''export THIRD_PARTY_INSTALL_PREFIX=%s;
+               export THIRD_PARTY_INSTALL_PREFIX_ARCH=%s;
+               cmake %s && make install VERBOSE=1 ''' % (log_build_dir, log_build_arch_dir, config.sdl_clone_dir))
+
+
+@task(task_class=TaskWithConfig)
+def build_atf(rewrite=False):
+    """
+    Build ATF
+    :param rewrite: if True, delete all build atf files and build again
+    """
+    with open_dir(config.work_dir), cd(config.atf_build_dir):
+        if rewrite:
+            run("git reset --hard && git clean -dfx")
+        qmake_path = config.qt_path + "/bin/qmake"
+        qmake_lib = config.qt_path + "/lib/"
+        run('''export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:%s; export QMAKE=%s; make''' % (
+            qmake_lib, qmake_path))
+
+
+@task(task_class=TaskWithConfig)
```

Comment:
@malirod I thin that most of functions should be tasks, because it is useful! Many time I want to run scripts again and don't want to build sdl ot ATF. or I want to use another version of SDL and don't want to rebuild atf .... etc.
All not task functions moved to utils.py file